adapters/repos/db/vector/flat/index.go

```diff
@@ -673,7 +687,12 @@ func (index *flat) UpdateUserConfig(updated schemaConfig.VectorIndexConfig, call
 }
 
 func (index *flat) Drop(ctx context.Context) error {
-	// nothing to do here
+	if index.metadata != nil {
+		if err := index.metadata.Close(); err != nil {
+			return err
+		}
+		os.Remove(filepath.Join(index.rootPath, index.getMetadataFile()))
```

Comment:
I would add a metadata.Remove() method and do the file removal there - we create the file there and should also remove it in there

Reply:
Added `index.removeMetadataFile`